it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,306 @@
+        }
+        // TODO: BQ profile builder
+
+        CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                .setParent(LocationName.of(projectId, location).toString())
+                .setConnectionProfile(connectionProfileBuilder)
+                .setConnectionProfileId(connectionProfileId)
+                .build();
+
+        ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+        createdConnectionProfileIds.add(connectionProfileId);
+
+        return reference;
+    }
+
+    public SourceConfig buildSourceConfig(String sourceConnectionProfileId) {
+        SourceConfig.Builder sourceConfigBuilder = SourceConfig
+                .newBuilder()
+                .setSourceConnectionProfile(ConnectionProfileName
+                        .format(projectId, location, sourceConnectionProfileId));
+
+        if (this.source == Source.MYSQL) {
+            MysqlRdbms.Builder mysqlRdmsBuilder = MysqlRdbms.newBuilder();
+            for (String db : this.allowedTables.keySet()) {
+                MysqlDatabase.Builder mysqlDbBuilder = MysqlDatabase.newBuilder().setDatabase(db);
+                for (String table : this.allowedTables.get(db)) {
+                    mysqlDbBuilder.addMysqlTables(MysqlTable.newBuilder().setTable(table));
+                }
+                mysqlRdmsBuilder.addMysqlDatabases(mysqlDbBuilder);
+            }
+            sourceConfigBuilder.setMysqlSourceConfig(MysqlSourceConfig.newBuilder().setIncludeObjects(mysqlRdmsBuilder));
+        }
+        // TODO: Oracle & Postgres source config builders
+
+        return sourceConfigBuilder.build();
+    }
+
+    public DestinationConfig buildDestinationConfig(String destinationConnectionProfileId) {
+        DestinationConfig.Builder destinationConfigBuilder = DestinationConfig
+                .newBuilder()
+                .setDestinationConnectionProfile(ConnectionProfileName
+                        .format(projectId, location, destinationConnectionProfileId));
+
+        if (this.destination == Destination.GCS) {
+            GcsDestinationConfig.Builder gcsDestConfigBuilder = GcsDestinationConfig
+                    .newBuilder().setPath(this.gcsRootPath);
+            if (this.destinationOutputFormat == AVRO_FILE_FORMAT) {
+                gcsDestConfigBuilder.setAvroFileFormat(AvroFileFormat.getDefaultInstance());
+            } else {
+                gcsDestConfigBuilder.setJsonFileFormat(JsonFileFormat.getDefaultInstance());
+            }
+            destinationConfigBuilder.setGcsDestinationConfig(gcsDestConfigBuilder);
+        }
+        // TODO: BQ Destination Builder
+
+        return destinationConfigBuilder.build();
+    }
+
+    @Override
+    public Stream createStream(String streamId,
+                               String sourceConnectionProfileId,
+                               String destinationConnectionProfileId)
```

Comment:
Based on my other comments, I think this method should take in a `SourceConfig` and `DestinationConfig` that were created using the other methods above. So the workflow from the caller would look something like...
```
// Create resource manager
DatastreamResourceManager rm = DefaultDatastreamResourceManager.builder(...).build();

// Create profiles
createSourceConnectionProfile(...);
createDestinationConnectionProfile(...);

// Create configs
SourceConfig source = rm.buildSourceConfig(...);
DestinationConfig dest = rm.buildDestinationConfig(...);

// Create and start stream
rm.createStream(..., source, dest);
rm.startStream(...);

// Wait...

// Close stream
rm.pauseStream(...);
rm.cleanupAll();
```